.github/workflows/release_deploy_pypi.yml

```diff
@@ -1,8 +1,41 @@
+name: "Release to PyPi"
+
 on:
-  workflow_dispatch:
+  release:
+    types: [published]
 
 jobs:
-  test:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/application-settings
+
+    permissions:
+      id-token: write
+
     steps:
-      - name: "Temp"
-        run: echo "Temporary, too be implemented"
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.2.2
```

Comment:
Why this version? @ viro we had a specific problem with this one, related to installation under windows so not that important for this workflow. But in general I think that it is worthwhile to update, as many fixes are made (https://github.com/python-poetry/poetry/blob/master/CHANGELOG.md)

Reply:
I copied the version of the action from the other workflow files, I'll update those. I actually expected that the dependabot should check that?